setup.py

```diff
@@ -117,12 +123,21 @@ def __init__(self):
         super(Protobuf, self).__init__()
         # TODO: allow user specify protobuf include_dirs libraries with flags
         use_conda = os.getenv('CONDA_PREFIX') and platform.system() == 'Windows'
+		# Find the path by finding protoc and suppose it's in ${PROTOBUF_PATH}/bin
+        proto_path = which('protoc')
+        if proto_path:
+            proto_path = os.path.realpath(proto_path + "/../")
+
 
         libs = []
+        link_args = []
         if os.getenv('PROTOBUF_LIBDIR'):
             libs.append(os.path.join(os.getenv('PROTOBUF_LIBDIR'), "libprotobuf"))
         elif use_conda:
             libs.append(os.path.join(os.getenv('CONDA_PREFIX'), "Library", "lib", "libprotobuf"))
+        elif proto_path:
+            libs.append(os.path.join(proto_path, "lib", "libprotobuf"))
+            link_args.extend(["-L" + os.path.join(proto_path, "lib"), "-lprotobuf"])
```

Comment:
1. `libprotobuf` has been added to `libs` already, why add `-lprotobuf` again here?
2. `-L` + `-l` is not enough. This only makes libprotobuf can be found at link time, but if libprotobuf is in some custom path, it won't be found by the loader.

Reply:
Yeah, actually, I don't know what dependency libs here does. when I print out the actual `gcc` compilation command, it doesn't seem to affect anything. That's why I added `-L` and `-l` which worked for me (checked by ldd). Any suggestions? Say adding `-Wl,-R`?